build-and-test.ps1

```diff
@@ -0,0 +1,46 @@
+[cmdletbinding()]
+param(
+    [string]$VersionFilter,
+    [string]$OSFilter,
+    [switch]$IsDryRun
+)
+
+Set-StrictMode -Version Latest
+$ErrorActionPreference = 'Stop'
+
+$(docker version) | ForEach-Object { Write-Host "$_" }
+
+$buildFilter = "*"
+if (-not [string]::IsNullOrEmpty($VersionFilter))
+{
+    $buildFilter = "$VersionFilter-$buildFilter"
+}
+if (-not [string]::IsNullOrEmpty($OSFilter))
+{
+    $buildFilter = "$buildFilter$OSFilter/*"
```

Comment:
`-OSFilter 1709` would match all 1709 Dockerfiles.  So one might expect `-OSFilter ltsc` would match all `ltsc`. But it wouldn't.  This is due to `2016` suffix, which means `-OSFilter ltsc2016` is how the parameter value should be for LTSC images.
Any preferences on the usage?

Reply:
Seems like there should be a `-` if an OSFilter is specified but not a VersionFilter - e.g. $buildFilter-$OSFilter/*.

I think it is just fine that `-OSFilter ltsc` is not supported.  The user can specify `*ltsc*` for this to work.

Reply:
`*ltsc*` sounds fine.

Hyphen `-` is not required. But would make it more accurate. I will update this as follows -

```
$buildFilter = "*"
if (-not [string]::IsNullOrEmpty($VersionFilter))
{
    $buildFilter = "$VersionFilter-$buildFilter"
}
if (-not [string]::IsNullOrEmpty($OSFilter))
{
    if ([string]::IsNullOrEmpty($VersionFilter))
    {
        $buildFilter = "$buildFilter-$OSFilter/*"
    }
    else
    {
        $buildFilter = "$buildFilter$OSFilter/*"
    }
}
```